resources/views/layouts/partials/header/autocomplete.blade.php

```diff
@@ -1,49 +1,65 @@
 
-<label for="autocomplete-input" class="sr-only">@lang('Search')</label>
-<input
-    id="autocomplete-input"
-    placeholder="@lang('Search')"
-    class="{{ $inputClasses }}"
-    v-on:focus="$root.loadAutocomplete = true"
-    v-if="!$root.loadAutocomplete"
->
+<div v-if="!$root.loadAutocomplete" class="relative w-full">
+    <label for="autocomplete-input" class="sr-only">@lang('Search')</label>
+    <input
+        id="autocomplete-input"
+        placeholder="@lang('Search')"
+        class="{{ $inputClasses }}"
+        v-on:focus="$root.loadAutocomplete = true"
+    >
+    <div class="rounded-r-xl border-l absolute right-0 inset-y-0 w-14 pointer-events-none flex items-center justify-center {{ config('rapidez.frontend.z-indexes.header-dropdowns') }}">
+        <x-heroicon-o-magnifying-glass class="size-5 text-neutral" />
+    </div>
+</div>
 
 <autocomplete
-    v-if="$root.loadAutocomplete"
+    v-else
     v-on:mounted="() => window.document.getElementById('autocomplete-input').focus()"
     v-bind:additionals="{{ json_encode(config('rapidez.frontend.autocomplete.additionals')) }}"
     v-bind:debounce="{{ config('rapidez.frontend.autocomplete.debounce') }}"
     v-bind:size="{{ config('rapidez.frontend.autocomplete.size') }}"
-    class="w-full"
+    class="relative w-full"
     v-cloak
 >
-    <x-rapidez::reactive-base slot-scope="{ results, resultsCount, searchAdditionals, debounce, size, highlight }">
+    <x-rapidez::reactive-base slot-scope="{ results, resultsCount, searchAdditionals, overlay, showOverlay, debounce, size, highlight }">
+        <div
+            class="pointer-events-none fixed z-10 inset-0 cursor-pointer bg-black/40 opacity-0 transition duration-500"
+            :class="{ 'pointer-events-auto opacity-100': overlay, 'opacity-0 pointer-events-none': !overlay }"
+        ></div>
+        <div class="rounded-r-xl border-l absolute right-0 inset-y-0 w-14 pointer-events-none flex items-center justify-center {{ config('rapidez.frontend.z-indexes.lightbox') }}">
+            <x-heroicon-o-magnifying-glass class="size-5 text-neutral" />
+        </div>
         <data-search
             placeholder="@lang('Search')"
             v-on:value-selected="search"
             component-id="autocomplete"
             :inner-class="{ input: '{{ $inputClasses }}' }"
             {{-- These classes are only used when you come from a page with a product listing, --}}
             {{-- click on a link that leads to a 404 page and try to use the search there --}}
-            class="relative [&_*]:!m-0 [&_[isclearicon=]]:!mr-2 [&_.cancel-icon]:!fill-[#595959] [&_[groupposition=right]]:!absolute [&_[groupposition=right]]:!top-2/4 [&_[groupposition=right]]:!right-0 [&_[groupposition=right]]:!-translate-y-2/4"
+            class="relative [&_*]:!m-0"
```

Comment:
Did you test this case?

Reply:
I have tested the case local:
1. Created a link that leads to a 404 page.
2. Navigated to a page where a product listing is used. 
3. Clicked on the link that leads to a 404 page. 
4. Used the autocomplete to search. 

Tested running `yarn dev` & `yarn prod`